lib/spack/spack/build_systems/oneapi.py

```diff
@@ -179,6 +189,39 @@ class IntelOneApiLibraryPackage(IntelOneApiPackage):
 
     """
 
+    def openmp_libs(self):
+        """Supply LibraryList for linking OpenMP"""
+
+        # NB: Hunting down explicit library files may be the Spack way of
+        # doing things, but it is better to add the compiler defined option
+        # e.g. -fopenmp
+
+        # If other packages use openmp, then all the packages need to
+        # support the same ABI. Spack usually uses the same compiler
+        # for all the packages, but you can force it if necessary:
+        #
+        # e.g. spack install blaspp%oneapi@2024 ^intel-oneapi-mkl%oneapi@2024
+        #
+        if self.spec.satisfies("%intel") or self.spec.satisfies("%oneapi"):
+            libname = "libiomp5"
+        elif self.spec.satisfies("%gcc"):
+            libname = "libgomp"
+        else:
+            raise_lib_error("MKL OMP requires one of %gcc, %oneapi,or %intel")
+
+        # query the compiler for the library path
+        with self.compiler.compiler_environment():
+            omp_lib_path = Executable(self.compiler.cc)(
+                "--print-file-name", f"{libname}.{dso_suffix}", output=str
+            ).strip()
+        # if the compiler cannot find the file, it returns the input path
+        if not os.path.exists(omp_lib_path):
+            raise_lib_error(f"Cannot locate OpenMP library: {omp_lib_path}")
+
+        omp_libs = LibraryList(omp_lib_path)
+        tty.info(f"mkl requires openmp library: {omp_libs}")
```

Comment:
Nit, for consistency? Not a blocker.
```suggestion
            raise_lib_error("OneApi package with OpenMP requires one of %gcc, %oneapi,or %intel")

        # query the compiler for the library path
        with self.compiler.compiler_environment():
            omp_lib_path = Executable(self.compiler.cc)(
                "--print-file-name", f"{libname}.{dso_suffix}", output=str
            ).strip()
        # if the compiler cannot find the file, it returns the input path
        if not os.path.exists(omp_lib_path):
            raise_lib_error(f"Cannot locate OpenMP library: {omp_lib_path}")

        omp_libs = LibraryList(omp_lib_path)
        tty.info(f"OneApi requires OpenMP library: {omp_libs}")
```